src/t_stream.c

```diff
@@ -1309,6 +1334,27 @@ int streamParseStrictIDOrReply(client *c, robj *o, streamID *id, uint64_t missin
     return streamGenericParseIDOrReply(c,o,id,missing_seq,1);
 }
 
+/* Helper for parsing a stream ID that is a range query interval. When the
+ * exclude argument is NULL, streamParseIDOrReply() is called and the interval
+ * is treated as close (inclusive). Otherwise, the exclude argument is set if 
+ * the interval is open (the "(" prefix) and streamParseStrictIDOrReply() is
+ * called in that case.
+ */
+int streamParseIntervalIDOrReply(client *c, robj *o, streamID *id, int *exclude, uint64_t missing_seq) {
+    char *p = o->ptr;
+    size_t len = sdslen(p);
+    int invalid = 0;
+    
+    if (exclude != NULL) *exclude = (len > 1 && p[0] == '(');
+    if (exclude != NULL && *exclude) {
+        robj *t = createStringObject(p+1,len-1);
+        invalid = (streamParseStrictIDOrReply(c,t,id,missing_seq) == C_ERR);
+        decrRefCount(t);
+    } else invalid = (streamParseIDOrReply(c,o,id,missing_seq) == C_ERR);
+    if (invalid) return C_ERR;
```

Comment:
maybe a matter of personal taste.
i think it's not nice that the `if` block has indented code, and the `else` is single line.
some would even say that if the `if` opens a curly braces block, the else needs to do that too.
besides that, i prefer statements that change the execution flow (`return` in this case) not to be in the same line of the `if`
```suggestion
    } else
        invalid = (streamParseIDOrReply(c,o,id,missing_seq) == C_ERR);
    if (invalid)
        return C_ERR;
```

Reply:
+1